test_scripts/PROJECTION/common.lua

```diff
@@ -0,0 +1,630 @@
+--]]
+local function tableToJsonFile(pTbl, pFileName)
+  local f = io.open(pFileName, "w")
+  f:write(json.encode(pTbl))
+  f:close()
+end
+
+--[[ @updatePTU: update PTU table with additional functional group for Navigation RPCs
+--! @parameters:
+--! pTbl - PTU table
+--! pAppId - application number (1, 2, etc.)
+--]]
+function m.updatePTU(pTbl, pAppId)
+  pTbl.policy_table.app_policies[m.getAppID(pAppId)] = {
+    keep_context = false,
+    steal_focus = false,
+    priority = "NONE",
+    default_hmi = "NONE",
+    groups = { "Base-4", "Location-1" }
+  }
+end
+
+--[[ @ptu: perform policy table update
+--! @parameters:
+--! pPTUpdateFunc - additional function for update
+--! pAppId - application number (1, 2, etc.)
+--]]
+local function ptu(pPTUpdateFunc, pAppId)
+  if not pAppId then pAppId = 1 end
+  local pts_file_name = commonFunctions:read_parameter_from_smart_device_link_ini("SystemFilesPath") .. "/"
+    .. commonFunctions:read_parameter_from_smart_device_link_ini("PathToSnapshot")
+  local ptu_file_name = os.tmpname()
+  local requestId = test.hmiConnection:SendRequest("SDL.GetURLS", { service = 7 })
+  EXPECT_HMIRESPONSE(requestId)
+  :Do(function()
+      test.hmiConnection:SendNotification("BasicCommunication.OnSystemRequest",
+        { requestType = "PROPRIETARY", fileName = pts_file_name })
+      getPTUFromPTS(ptuTable)
+
+      m.updatePTU(ptuTable, pAppId)
+
+      if pPTUpdateFunc then
+        pPTUpdateFunc(ptuTable)
+      end
+
+      tableToJsonFile(ptuTable, ptu_file_name)
+
+      local event = events.Event()
+      event.matches = function(e1, e2) return e1 == e2 end
+      EXPECT_EVENT(event, "PTU event")
+
+      local function getAppsCount()
+        local count = 0
+        for _ in pairs(hmiAppIds) do
+          count = count + 1
+        end
+        return count
+      end
+      for id = 1, getAppsCount() do
+        local mobileSession = m.getMobileSession(id)
+        mobileSession:ExpectNotification("OnSystemRequest", { requestType = "PROPRIETARY" })
+        :Do(function()
+            print("App ".. id .. " was used for PTU")
+            RAISE_EVENT(event, event, "PTU event")
+            local corIdSystemRequest = mobileSession:SendRPC("SystemRequest",
+              { requestType = "PROPRIETARY" }, ptu_file_name)
+            EXPECT_HMICALL("BasicCommunication.SystemRequest")
+            :Do(function(_, d3)
+                test.hmiConnection:SendResponse(d3.id, "BasicCommunication.SystemRequest", "SUCCESS", { })
+                test.hmiConnection:SendNotification("SDL.OnReceivedPolicyUpdate", { policyfile = d3.params.fileName })
+              end)
+            mobileSession:ExpectResponse(corIdSystemRequest, { success = true, resultCode = "SUCCESS" })
+            :Do(function() os.remove(ptu_file_name) end)
+          end)
+        :Times(AtMost(1))
+      end
+    end)
+end
+
+--[[ @allowSDL: sequence that allows SDL functionality
+--! @parameters: none
+--]]
+local function allowSDL()
```

Comment:
@GetmanetsIrina Please update this function as it's done in https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/692dabdf4ddad61381bd599017749dcefe52d963/test_scripts/RC/commonRC.lua#L126

Reply:
Locally changed.